src/query/service/src/sessions/query_ctx.rs

```diff
@@ -354,6 +359,11 @@ impl TableContext for QueryContext {
     fn get_processes_info(&self) -> Vec<ProcessInfo> {
         SessionManager::instance().processes_info()
     }
+
+    // Get Sideload Options.
+    fn get_sideload(&self) -> Option<SideloadOptions> {
+        self.shared.get_sideload()
+    }
```

Comment:
can this method only placed in `impl QueryContext` instead of `impl TableContext for QueryContext`?

IMHO, the sideload options makes sense in QueryContext (it takes effect in a query's lifetime), but it may hard to understand as a property in the table context.

Reply:
The `Binder` only have a `TableContext` when binding statement, I think the name `TableContext` is somehow missleading.

Reply:
It seems @Xuanwo changed this previously https://github.com/datafuselabs/databend/pull/7600.
Can you help provide some background for this?

Reply:
I added this to address circle dependences. `Planner` can't relay on `QueryContext` directly.

Reply:
How about name it as BinderContext ?

Reply:
Maybe we can rename it in next PR.

Reply:
https://github.com/datafuselabs/databend/issues/9263